### PR TITLE
feat: atlas-basic update

### PR DESCRIPTION
**Pull request by rubenVB01**
## Description

In this PR I did the following changes:
- Updated the module so that now it uses the use_existing_project flag (so that users can provide an existing project), and adapted the code to do so
- Added a test for that use case
- Added an example for that use case
- Updated the documentation taking into account the feedback from Zuhair and Kailie (who is in charge of doing the guide that will be published for new users)

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -13,28 +13,34 @@ locals {
     }])
     scopes = tolist([])
   }])
-  database_users = length(var.database_users) == 0 ? local.database_users_default : var.database_users
+  database_users = length(var.database_users) == 0 && !var.use_existing_project ? local.database_users_default : var.database_users
```
Comment (EspenAlbert) [inline main.tf:16]: curious about this change?

**Review by EspenAlbert — APPROVED** [review]
Nice! Like the data source approach and depending on the resource.
We might need some changes to the tests. It looks like it is failing

File: main.tf (hunk 1 above)
Comment (rubenVB01) [reply main.tf:16]: Yes, I should have given a bit of context. I found that the following problem, imagine this scenario:
A user calls the module with an existing project and does not provide a database user (the module is going to create a default database user).
Then, the user calls the module again for the same project and again does not provide a database user. The problem here is that the module is going to try to create again a default database user (with the same name as the previous), which gives an error or is something we don't want to happen.
The work-around I decided to do is to only create a default database user if the module creates the project and no user is provided. If an existing project is provided, no default user will be created even if no user is provided.

File: tests/atlas_basic_empty.tftest.hcl [hunk 2]
```diff
@@ -15,7 +15,7 @@ run "atlas_basic_empty" {
   }
 
   assert {
-    condition     = mongodbatlas_project.project.name == "terraform-mongodbatlas-atlas-basic"
+    condition     = mongodbatlas_project.project[0].name == "terraform-mongodbatlas-atlas-basic"
```
Comment (AgustinBettati) [inline tests/atlas_basic_empty.tftest.hcl:18]: curious why this change was needed

**Review by AgustinBettati — APPROVED** [review]
LGTM, great improvement to the module with a simple implementation

File: variables.tf [hunk 3]
```diff
@@ -10,6 +10,12 @@ variable "project_name" {
   default     = "terraform-mongodbatlas-atlas-basic"
 }
 
+variable "use_existing_project" {
+  description = "Flag that determines whether the module uses an existing project or creates a new one."
+  type        = bool
+  default     = false
+}
+
 # mongodbatlas_project_ip_access_list
```
Comment (AgustinBettati) [inline variables.tf:19]: should we adjust docs on database_users to clarify that default value will not be present when using existing project?

File: tests/atlas_basic_empty.tftest.hcl (hunk 2 above)
Comment (rubenVB01) [reply tests/atlas_basic_empty.tftest.hcl:18]: because now, if a project resource is created, it is created as a list of 1 element because of the condition:

`count  = var.use_existing_project ? 0 : 1`

File: variables.tf (hunk 3 above)
Comment (rubenVB01) [reply variables.tf:19]: Added it on the global readme. However I could also add it in the description of the variable if its necessary / clearer

Comment (AgustinBettati) [reply variables.tf:19]: Great, yes I think that would be good so we cover the case of users that go directly to docs of the input variables.
